### DISCUSSION TITLE
Change DataGrid row background based on current item.Status

### DISCUSSION THREAD

**Discussion by sn4k3** [AvaloniaUI/Avalonia#17788] [Q&A] [answered]
I would like to change the background of entire row based on row `item.Status`, to be Red when is failed and null when success. Due this I can't use the one time `OnRowLoading` event. Is there any alternative?

![image](https://github.com/user-attachments/assets/37e467b7-2b52-4f21-8048-ee1a74fd4acb)

If not possible I think the alternative would be set the text in Red. Are two `TextBlock` with different classes and IsVisible the best way to do the class change?

**Comment by msneijders** [comment, marked as answer]
This could do it:

```xaml
<DataGrid ...>
    <DataGrid.RowTheme>
        <ControlTheme
            x:DataType="vm:YourItemType"
            BasedOn="{StaticResource {x:Type DataGridRow}}"
            TargetType="DataGridRow">
            <Setter Property="Background" Value="{Binding Status, Converter={StaticResource StatusToRowColorConverter}}" />
        </ControlTheme>
    </DataGrid.RowTheme>
</DataGrid>
```

Docs about making a custom converter [here](https://docs.avaloniaui.net/docs/guides/data-binding/how-to-create-a-custom-data-binding-converter). (I made up `StatusToRowColorConverter`, you'd have to implement that yourself)

If your viewmodel object is observable and property `Status` raises a `PropertyChanged` the DataGridRow will respond to it.

**Reply by sn4k3** [reply]
The `BasedOn` is giving trouble

![image](https://github.com/user-attachments/assets/7c037651-9564-4389-8083-173757ac7dda)

**Reply by msneijders** [reply]
I tested this xaml in avalonia's ControlCatalog app, in there it works in runtime. You are including a base ControlTheme for `DataGridRow` in your resources?

**Reply by sn4k3** [reply]
That was it, I was not including DataGrid style because i'm using an external UI theme, looks like they lack that resource, I will check.
Applying default theme worked with your sample, thank you!
